### ISSUE TITLE
Simply mount options for uid/gid

### ISSUE THREAD

**Issue by msau42** [GoogleCloudPlatform/gcs-fuse-csi-driver#16]
Right now users have to set the driver's mount options to match the uid/gid settings.

Is UID required, or is gid/fsgroup sufficient? Normally for pod volumes, having the volume ownership match the fsgroup (ie supplemental group of the contaner) is enough and uid for the volume doesn't matter.

If you enable [delegate fsgroup to csidriver](https://kubernetes-csi.github.io/docs/support-fsgroup.html#delegate-fsgroup-to-csi-driver), then k8s will send over the fsgroup id to the NodePublishVolume call, so that could eliminate needing to repeat gid in the mount options.

**Comment by msau42** [comment]
Also related, evaluate [fsGroupPolicy](https://kubernetes-csi.github.io/docs/support-fsgroup.html#csi-volume-fsgroup-policy) as well.

**Comment by songjiaxun** [comment]
The driver supports [Delegate fsGroup to CSI Driver](https://kubernetes-csi.github.io/docs/support-fsgroup.html#delegate-fsgroup-to-csi-driver) after the [commit](https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver/commit/1c9e63c12d404ace3f8cea68a39afd1619b6cab9). This will be included in the next release.

With this feature, users do not need to explicitly specify `uid` or `gid` on the gcsfuse mount options. They can just specify `Pod.spec.securityContext.fsGroup`, and the `fsGroup` id will be automatically passed to gcsfuse `gid` flag. Meanwhile, the `file-mode` and `dir-mode` will be set as `664` and `775` respectively. Users can still overwrite these flags, meaning the exiting workloads won't be affected.

**Comment by zevisert** [comment]
I haven't tested yet, but isn't this closed by [release 0.1.7](https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver/releases/tag/v0.1.7)?

**Comment by zevisert** [comment]
I have a `PodTemplateSpec.spec.securityContext.fsGroup` and I still see 
```
gcsfuse mounting with args [gcsfuse --uid 0 --gid 0 --temp-dir /gcsfuse-tmp/...
```

Is there a mount option or annotation I need to provide to use this, or is it simply not rolled out to the GKE-managed version yet?

**Comment by songjiaxun** [comment]
I can confirm that `1.27.3-gke.100` does not container the new feature. I am updating the release note, from where you can see which GKE version contains the new feature. The documentation will be updated shortly.

**Comment by songjiaxun** [comment]
The doc is updated to describe which GKE version includes the feature: https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver/issues/16

@zevisert  This feature was included in CSI version `v0.1.7`. As of today, the version `1.27.8-gke.1067000` includes the feature.
